### PR TITLE
Js/da 274 part b

### PR DESCRIPTION
**Pull request by johnstill**
Removes a bunch of dead code and simplifies the configuration implementation / interface.

### CODE REVIEW

**Comment by johnstill** [comment]
I'm not seeing the circle ci failure locally; any suggestions are welcome.

**Comment by Seeker1911** [comment]
Looks the same as the problem I had importing backoff. Not sure if circle-ci runs setup_env.

File: rest-api/config_api.py [hunk 1]
```diff
@@ -64,45 +63,60 @@ def check_config_admin():
...
+    if config.getSettingJson(config.ALLOW_NONPROD_REQUESTS, False):
+      date = request.headers.get('x-pretend-date', None)
+      if date:
+        date = parse_date(date)
+
+    client_id = app_util.get_oauth_id()
+    config.store(model, date=date, client_id=client_id)
+    return self.make_response_for_resource(model.configuration)
+
+  def make_response_for_resource(self, data):
```
Comment (danqrodney) [inline rest-api/config_api.py]: You can delete this whole thing and just return model.configuration above. Configs don't have last_modified. (The purpose of this logic was for e-tag checking, but we don't do that on configs, either.)

File: rest-api/config_api.py [hunk 2]
```diff
@@ -64,45 +63,60 @@ def check_config_admin():
...
+    data = model.configuration or {}
+    return self.make_response_for_resource(data)
 
-  def put(self, key=None):
-    return super(ConfigApi, self).put(key or config.CONFIG_SINGLETON_KEY)
-
-  def validate_object(self, obj, a_id=None):
-    super(ConfigApi, self).validate_object(obj, a_id)
-    if obj.key.id() != config.CONFIG_SINGLETON_KEY:
+  def post(self, key=config.CONFIG_SINGLETON_KEY):
```
Comment (danqrodney) [inline rest-api/config_api.py]: Seems like post and put should share the same implementation,differing only by the lookup for existing config?

File: rest-api/config.py [hunk 3]
```diff
@@ -46,86 +48,80 @@
...
-      initialize_config()
-      obj = super(ConfigurationDAO, self).load_if_present(id_, ancestor_id)
-    return obj
+class ConfigurationHistory(ndb.Model):
+  date = ndb.DateTimeProperty(auto_now_add=True)
+  obj = ndb.StructuredProperty(Configuration, repeated=False)
+  client_id = ndb.StringProperty()
+
+
+def load(_id, date=None):
```
Comment (danqrodney) [inline rest-api/config.py]: We need tests of load and store to replace data_access_object_test tests below.

File: rest-api/config_api.py (hunk 1 above)
Comment (johnstill) [reply rest-api/config_api.py]: Nice! I wasn't sure if a stored config might have a last modified tag or not.

File: rest-api/config_api.py (hunk 2 above)
Comment (johnstill) [reply rest-api/config_api.py]: not quite - `put` also does some stuff with `date` and `client_id`.  They are similar though; I've managed to reduce them a little more.

**Pull request merged**